slick/src/main/scala/slick/jdbc/HsqldbProfile.scala

```diff
@@ -125,8 +128,160 @@ trait HsqldbProfile extends JdbcProfile {
+      }
+    }
+
+    class OffsetTimeJdbcType extends super.OffsetTimeJdbcType with HsqldbTimeJdbcTypeWithOffset {
+
+      override def sqlType = java.sql.Types.TIME_WITH_TIMEZONE
+
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "TIME(9) WITH TIME ZONE"
+
+      override val hasLiteralForm: Boolean = false
+
+      override def setValue(v: OffsetTime, p: PreparedStatement, idx: Int) = {
+        @inline val correctedOffsetTime: String = offsetConvertISOToHsqldb(v.format(timeFormatter))
+        p.setString(idx, correctedOffsetTime)
+      }
+
+      override def updateValue(v: OffsetTime, r: ResultSet, idx: Int) = {
+        @inline val correctedOffsetTime: String = offsetConvertISOToHsqldb(v.format(timeFormatter))
+        r.updateString(idx, correctedOffsetTime)
+      }
+
+      override def getValue(r: ResultSet, idx: Int): OffsetTime = {
+        r.getString(idx) match {
+          case null => null
+          case hsqldbString =>
+            @inline val normalizedIsoString : String = offsetConvertHsqldbToISO(hsqldbString)
+            // in a subsecond greater than 00:00:00 and with an offset, we seem to get back a string like
+            // '24:00:00.745+01:00' instead of '00:00:00.745+01:00' which is an invalid OffsetTime,
+            // so update the leading 24 to 00
+            OffsetTime.parse(
+              if (normalizedIsoString.startsWith("24:00:00"))
+                "00" + normalizedIsoString.substring(2)
+              else
+                normalizedIsoString, timeFormatter)
+        }
+      }
+    }
+
+    class OffsetDateTimeJdbcType extends super.OffsetDateTimeJdbcType with HsqldbTimeJdbcTypeWithOffset {
+
+      override def sqlType = java.sql.Types.TIMESTAMP_WITH_TIMEZONE
+
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "TIMESTAMP(9) WITH TIME ZONE"
+
+      override val hasLiteralForm: Boolean = false
+
+      override def setValue(v: OffsetDateTime, p: PreparedStatement, idx: Int) = {
+        @inline val correctedOffsetTime: String = offsetConvertISOToHsqldb(v.format(datetimeFormatter))
+        p.setString(idx, correctedOffsetTime)
+      }
+
+      override def updateValue(v: OffsetDateTime, r: ResultSet, idx: Int) = {
+        @inline val correctedOffsetTime: String = offsetConvertISOToHsqldb(v.format(datetimeFormatter))
+        r.updateString(idx, correctedOffsetTime)
+      }
+
+      override def getValue(r: ResultSet, idx: Int): OffsetDateTime = {
+        r.getString(idx) match {
+          case null => null
+          case hsqldbString =>
+            @inline val normalizedIsoString: String = offsetConvertHsqldbToISO(hsqldbString)
+            OffsetDateTime.parse(normalizedIsoString,datetimeFormatter)
+        }
+      }
+    }
+    class InstantJdbcType extends super.InstantJdbcType with HsqldbTimeJdbcTypeWithOffset {
+      override def sqlType = java.sql.Types.TIMESTAMP_WITH_TIMEZONE
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "TIMESTAMP(9) WITH TIME ZONE"
```

Comment:
same comment as with the other Instance types... no zone needed if we expect UTC as implied.